File: TemplateVB/Addin/AddinRibbon.vb
Imports System.Drawing
Imports System.Runtime.InteropServices
Imports Microsoft.Office.Core


''' 
''' User interface manager for Visio 2010 and above
''' Creates and controls ribbon UI
''' 
$if$ ($vstoAddin$ == true)<ComVisible(True)>
$endif$Public Partial Class $if$ ($vstoAddin$ == true)AddinUI$else$ThisAddIn$endif$
    Implements IRibbonExtensibility
	Private _ribbon As Microsoft.Office.Core.IRibbonUI

#Region "IRibbonExtensibility Members"

    Public Function GetCustomUI(ribbonId As String) As String Implements IRibbonExtensibility.GetCustomUI
        Return My.Resources.Ribbon
    End Function

#End Region

#Region "Ribbon Callbacks"
    
    Public Function IsRibbonCommandEnabled(ctrl As Microsoft.Office.Core.IRibbonControl) As Boolean
            Return $thisAddIn$IsCommandEnabled(ctrl.Id)
    End Function
    
    Public Function IsRibbonCommandChecked(ctrl As Microsoft.Office.Core.IRibbonControl) As Boolean
            Return $thisAddIn$IsCommandChecked(ctrl.Id)
    End Function
    
    Public Sub OnRibbonButtonCheckClick(control As Microsoft.Office.Core.IRibbonControl, pressed As Boolean)
            $thisAddIn$OnCommand(control.Id)
    End Sub
    
    Public Sub OnRibbonButtonClick(control As Microsoft.Office.Core.IRibbonControl)
            $thisAddIn$OnCommand(control.Id)
    End Sub
    
    Public Function OnGetRibbonLabel(control As Microsoft.Office.Core.IRibbonControl) As String
            Return $thisAddIn$GetCommandLabel(control.Id)
    End Function
    
    Public Sub OnRibbonLoad(ribbonUI As Microsoft.Office.Core.IRibbonUI)
        _ribbon = ribbonUI
    End Sub
    
    Public Function GetRibbonImage(control As Microsoft.Office.Core.IRibbonControl) As Bitmap
            Return $thisAddIn$GetCommandBitmap(control.Id)
    End Function
    
    #End Region
    
    Public Sub UpdateRibbon()
        _ribbon.Invalidate()
    End Sub

End Class

File: TemplateCS/Addin/AddinRibbon.cs
using System.Drawing;
using System.Runtime.InteropServices;
using Office = Microsoft.Office.Core;

namespace $csprojectname$
{
    /// <summary>
    /// User interface manager for Visio 2010 and above
    /// Creates and controls ribbon UI
    /// </summary>
    /// 

    $if$ ($vstoAddin$ == true)[ComVisible(true)]
    $endif$public partial class $if$ ($vstoAddin$ == true)AddinUI$else$ThisAddIn$endif$ : Office.IRibbonExtensibility
    {
        private Office.IRibbonUI _ribbon;

        #region IRibbonExtensibility Members

        public string GetCustomUI(string ribbonId)
        {
            return Properties.Resources.Ribbon;
        }

        #endregion

        #region Ribbon Callbacks

        public bool IsRibbonCommandEnabled(Office.IRibbonControl ctrl)
        {
            return $thisAddIn$IsCommandEnabled(ctrl.Id);
        }

        public bool IsRibbonCommandChecked(Office.IRibbonControl ctrl)
        {
            return $thisAddIn$IsCommandChecked(ctrl.Id);
        }

        public void OnRibbonButtonCheckClick(Office.IRibbonControl control, bool pressed)
        {
            $thisAddIn$OnCommand(control.Id);
        }

        public void OnRibbonButtonClick(Office.IRibbonControl control)
        {
            $thisAddIn$OnCommand(control.Id);
        }

        public string OnGetRibbonLabel(Office.IRibbonControl control)
        {
            return $thisAddIn$GetCommandLabel(control.Id);
        }

        public void OnRibbonLoad(Office.IRibbonUI ribbonUI)
        {
            _ribbon = ribbonUI;
        }

        public Bitmap GetRibbonImage(Office.IRibbonControl control)
        {
            return $thisAddIn$GetCommandBitmap(control.Id);
        }

        #endregion

        public void UpdateRibbon()
        {
            _ribbon.Invalidate();
        }
    }
}
